vioscsi/vioscsi.c

```diff
@@ -438,60 +438,189 @@ VioScsiFindAdapter(
 
 ENTER_FN();
 
+    /* adaptExt: struct ADAPTER_EXTENSION
+     * Contains driver specific state and runtime information. This struct is
+     * created and managed by the driver and not by Microsoft/Storport.
+     */
     adaptExt = (PADAPTER_EXTENSION)DeviceExtension;
     RtlZeroMemory(adaptExt, sizeof(ADAPTER_EXTENSION));
 
     adaptExt->dump_mode  = IsCrashDumpMode;
     adaptExt->hba_id     = HBA_ID;
-    ConfigInfo->Master                      = TRUE;
-    ConfigInfo->ScatterGather               = TRUE;
-    ConfigInfo->DmaWidth                    = Width32Bits;
+
+    /* ConfigInfo: PORT_CONFIGURATION_INFORMATION
+     * Contains configuration information for a host bus adapter.
+     * Reference:
+     *   https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/storport/ns-storport-_port_configuration_information
+     */
+    ConfigInfo->AlignmentMask               = 0x3;
     ConfigInfo->Dma32BitAddresses           = TRUE;
     ConfigInfo->Dma64BitAddresses           = SCSI_DMA64_MINIPORT_FULL64BIT_SUPPORTED;
-    ConfigInfo->WmiDataProvider             = TRUE;
-    ConfigInfo->AlignmentMask               = 0x3;
-    ConfigInfo->MapBuffers                  = STOR_MAP_NON_READ_WRITE_BUFFERS;
-    ConfigInfo->SynchronizationModel        = StorSynchronizeFullDuplex;
+    ConfigInfo->DmaWidth                    = Width32Bits;
     ConfigInfo->HwMSInterruptRoutine        = VioScsiMSInterrupt;
     ConfigInfo->InterruptSynchronizationMode=InterruptSynchronizePerMessage;
+    ConfigInfo->MapBuffers                  = STOR_MAP_NON_READ_WRITE_BUFFERS;
+    ConfigInfo->Master                      = TRUE;
+    ConfigInfo->ScatterGather               = TRUE;
+    ConfigInfo->SynchronizationModel        = StorSynchronizeFullDuplex;
+    ConfigInfo->WmiDataProvider             = TRUE;
 
+    /* Define entry points to the driver's WMI library callback routines. */
     VioScsiWmiInitialize(DeviceExtension);
 
+    /* Try to bring up the device. */
     if (!InitHW(DeviceExtension, ConfigInfo)) {
-        RhelDbgPrint(TRACE_LEVEL_FATAL, " Cannot initialize HardWare\n");
+        RhelDbgPrint(TRACE_LEVEL_FATAL, " Cannot initialize Hardware\n");
         return SP_RETURN_NOT_FOUND;
     }
 
-    /* Set num_queues and seg_max to some sane values, to keep "Static Driver Verification" happy */
+    /* Initialize num_queues and seg_max to temporary values to keep
+     * Static Driver Verification happy. These values will be immediately
+     * reconfigured within GetScsiConfig below, which will retrieve the runtime
+     * values advertised by the underlying device.
+     */
     adaptExt->scsi_config.num_queues = 1;
-    adaptExt->scsi_config.seg_max = MAX_PHYS_SEGMENTS + 1;
+    adaptExt->scsi_config.seg_max    = SCSI_MINIMUM_PHYSICAL_BREAKS;
```

Comment:
Perhaps `SCSI_MAXIMUM_PHYSICAL_BREAKS` would make this silly line looks more rational?